### PR TITLE
[Website] Migrate from FontAwesome 4.7.0 -> 5.8.2

### LINKED ISSUE #260: Updated font awesome
So I'm wanting to add some social icons to the footer but currently the site makes use of FA 4.7 which does not contain all the icons I am seeking to use.

Currently cdnjs doesn't serve a newer version (5.x) so the only options for me in this regard are to use FA direct from FA, or to wait for (or chase) the cdnjs update of the lib.

@PeterDaveHello Any thoughts regarding this?

### PR DESCRIPTION
**Pull request by MattIPv4**
This PR fixes #260
**Migrating the website from FontAwesome 4.7.0 to the newer 5.8.2**
 - Replace comments icon with correct Discourse icon in footer
 - Replace all other icons like for like

### CODE REVIEW

File: templates/layout.html [hunk 1]
```diff
@@ -100,10 +100,10 @@
     <footer class="footer">
       <div class="container">
         <p class="text-muted">
-          Donate CDNJS $5 on <a href="https://www.bountysource.com/teams/cdnjs" target="_blank">Bountysource</a> or become a contributor on <a href="https://github.com/cdnjs/cdnjs" target="_blank"><i class="fa fa-github"></i> GitHub</a> to make the project better and better!
+          Donate CDNJS $5 on <a href="https://www.bountysource.com/teams/cdnjs" target="_blank">Bountysource</a> or become a contributor on <a href="https://github.com/cdnjs/cdnjs" target="_blank"><i class="fab fa-github"></i> GitHub</a> to make the project better and better!
```
Comment (PeterDaveHello) [inline templates/layout.html]: Conflicts here, need to rebase with the latest master branch, thanks.

File: public/js/main.js [hunk 2]
```diff
@@ -243,10 +243,10 @@ function setFileURLs(new_provider) {
         var user = getSafeHighlightedValue(hit._highlightResult.github.user);
         var repo = getSafeHighlightedValue(hit._highlightResult.github.repo);
         githubDetails = '<ul class="list-inline">' +
-          '<li><i class="fa fa-github"></i> <a href="https://github.com/' + hit.github.user + '/' + hit.github.repo + '" target="_blank">' + user + '/' + repo + '</a></li>' +
-          '<li><i class="fa fa-eye"></i> ' + hit.github.subscribers_count + '</li>' +
-          '<li><i class="fa fa-star"></i> ' + hit.github.stargazers_count + '</li>' +
-          '<li><i class="fa fa-code-fork"></i> ' + hit.github.forks + '</li>' +
+          '<li><i class="fab fa-github"></i> <a href="https://github.com/' + hit.github.user + '/' + hit.github.repo + '" target="_blank">' + user + '/' + repo + '</a></li>' +
+          '<li><i class="far fa-eye"></i> ' + hit.github.subscribers_count + '</li>' +
```
Comment (PeterDaveHello) [inline public/js/main.js]: I wonder if the "Regular" style is okay for us? Looks like it's not part of free plan, I'm not sure, ref: https://fontawesome.com/how-to-use/on-the-web/referencing-icons/basic-use

File: templates/library.html [hunk 3]
```diff
@@ -9,17 +9,17 @@
     <h1 style="margin-top: 00px;" id="libraryName">{{library.name}}</h1>
     {{#watch}}
     <ul class="list-inline">
-      <li><i class="fa fa-eye"></i><strong> {{watch}}</strong></li>
-      <li><i class="fa fa-star"></i><strong> {{star}}</strong></li>
-      <li><i class="fa fa-code-fork"></i><strong> {{fork}}</strong></li>
+      <li><i class="far fa-eye"></i><strong> {{watch}}</strong></li>
+      <li><i class="fas fa-star"></i><strong> {{star}}</strong></li>
```
Comment (PeterDaveHello) [inline templates/library.html]: I suppose we should use same style on the same block?

File: public/js/main.js (hunk 2 above)
Comment (MattIPv4) [reply public/js/main.js]: https://fontawesome.com/icons/eye?style=regular the regular eye is free to use :) It's the shim for the fa 4.x eye.

File: templates/library.html (hunk 3 above)
Comment (MattIPv4) [reply templates/library.html]: | Solid | Regular |
|------|--------- |
| image | image |
| image | image |

I personally find the solid star to work better and the regular eye - the solid eye feels too bloated?

File: public/js/main.js (hunk 2 above)
Comment (PeterDaveHello) [reply public/js/main.js]: hmmm, thanks. Do you have any idea about "Pro Required" on the link I post?

Comment (MattIPv4) [reply public/js/main.js]: Most regular and light icons do fall under pro only, however, as the regular eye is the shim for FA 4.x, it is available for free use.

File: templates/library.html (hunk 3 above)
Comment (PeterDaveHello) [reply templates/library.html]: In this case, maybe align to solid style IMO, should make the project more maintainable, thanks.

File: public/js/main.js (hunk 2 above)
Comment (PeterDaveHello) [reply public/js/main.js]: Cool :+1:

**Comment by MattIPv4** [comment]
@PeterDaveHello Using consistent icon style now and rebased against latest master

File: public/js/main.js [hunk 4]
```diff
@@ -93,7 +93,7 @@ function setFileURLs(new_provider) {
           copyEl.children('ul').remove();
           copyEl.append('<ul class="dropdown-menu copy-options">' +
             '<li><a data-copy-type="https:" class="copy-https-url copy-button" href="javascript:void(0);">Copy Url</a></li>' + SRIcopyWithoutTagButton +
-              SRIcopyButton + 
+              SRIcopyButton +
```
Comment (PeterDaveHello) [inline public/js/main.js]: Let's prevent touching anything not related.

File: templates/layout.html [hunk 5]
```diff
@@ -21,7 +21,7 @@
     <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/pace/1.0.2/themes/orange/pace-theme-minimal.min.css" integrity="sha256-kb8pRNu1sIwQEWAO/Mqt1S5PZ5xiLd4nBMoSsqdxKPs=" crossorigin="anonymous" />
     <script src="https://cdnjs.cloudflare.com/ajax/libs/pace/1.0.2/pace.min.js" integrity="sha256-EPrkNjGEmCWyazb3A/Epj+W7Qm2pB9vnfXw+X6LImPM=" crossorigin="anonymous" async></script>
     <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/united/bootstrap.min.css" integrity="sha256-IYGI0ovdtvTnw19p8rG6jZjmlX+5x+4rw9uAhk+5OC4=" crossorigin="anonymous" />
-    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.8.2/css/all.min.css" integrity="sha256-BtbhCIbtfeVWGsqxk1vOHEYXS6qcvQvLMZqjtpWUEx8=" crossorigin="anonymous" />
```
Comment (PeterDaveHello) [inline templates/layout.html]: BTW, maybe update to the latest v5.9.0?

**Comment by PeterDaveHello** [comment]
The commit message is dated, just missed it, btw, the second line should always be blank, content should start from line 3.

**Pull request merged**